airflow/models/dagrun.py

```diff
@@ -1274,37 +1280,44 @@ def schedule_tis(self, schedulable_tis: Iterable[TI], session: Session = NEW_SES
                 schedulable_ti_ids.append((ti.task_id, ti.map_index))
 
         count = 0
+        max_tis_per_query = airflow_conf.getint("scheduler", "max_tis_per_query")
```

Comment:
Since this function is actually called by a Job object, which would have the `max_tis_per_query` attribute available, we should make the Job pass the value in as an argument instead of reading from config. The less magic global values, the better.

Reply:
It's a good idea, but the method `schedule_tis` is also called by `TaskInstance` in the method  `schedule_downstream_tasks` and it doesn't have the `max_tis_per_query` attribute.

Reply:
`schedule_downstream_tasks` is also only called by a job.

Reply:
Got it, I've added the changes

Reply:
I've faced a problem with getting `max_tis_per_query` in the `LocalTaskJobRunner`, in that class job might be Job or JobPydantic instance and JobPydantic doesn't have max_tis_per_query and the static check failed there, what do you suggest to me?

Reply:
Hmm, @potiuk want to hear your thoughts since this will likely be affected by your refactoring down the road.

Reply:
I've just added the max_tis_per_query parameter to the JobPydantic, I think it's gonna work

Reply:
@uranusjr Can I close conversation as resolved?